modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -482,7 +535,7 @@ else if (newKeys != null) {
     public void groupKey(int grpId, byte[] encGrpKey) {
         assert !grpEncKeys.containsKey(grpId);
 
-        Serializable encKey = getSpi().decryptKey(encGrpKey);
+        Serializable encKey = withMasterKeyChangeReadLock(() -> getSpi().decryptKey(encGrpKey));
```

Comment:
What is the MK will change after this line and before `encKey` will be written to the metastore?

Reply:
We do not encrypt group key here. Decrypt is under the read lock because of it can be at the same time with the local new master key check.